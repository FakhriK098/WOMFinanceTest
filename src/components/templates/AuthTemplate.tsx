import React from 'react'
import { View, StyleSheet, useColorScheme } from 'react-native'
import Text from '../atoms/Text'
import type { AuthTemplateProps } from '@typings/components'
import { colors } from '@theme/colors'
import { spacing } from '@theme/spacing'
import { radius } from '@theme/radius'
import { typography } from '@theme/typography'
import Spacer from '../atoms/Spacer'

const AuthTemplate = ({ title, children }: AuthTemplateProps) => {
  const isDark = useColorScheme() === 'dark'
  return (
    <View style={[styles.container, { backgroundColor: isDark ? colors.background.dark : colors.background.light }]}>
      <View style={styles.card}>
        <Text weight="bold" size={typography.xxl} color={isDark ? colors.text.dark : colors.text.light}>
          {title}
        </Text>
        <Spacer size={spacing.xl} />
        {children}
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: spacing.lg,
    alignItems: 'center',
    justifyContent: 'center',
  },
  card: {
    width: '100%',
    borderRadius: radius.lg,
    padding: spacing.lg,
    gap: spacing.sm,
  },
})

export default React.memo(AuthTemplate)
