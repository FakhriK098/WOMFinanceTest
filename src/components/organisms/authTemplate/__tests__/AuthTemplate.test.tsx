import React from 'react';
import renderer, { act } from 'react-test-renderer';
import * as RN from 'react-native';

import AuthTemplate from '../AuthTemplate';

describe('AuthTemplate organism', () => {
  it('matches snapshot (light) and renders title/children', () => {
    jest.spyOn(RN, 'useColorScheme').mockReturnValue('light');
    let tr: renderer.ReactTestRenderer;
    act(() => {
      tr = renderer.create(
        <AuthTemplate title="Welcome">
          <></>
        </AuthTemplate>,
      );
    });
    expect(tr!.toJSON()).toMatchSnapshot();
    const title = tr!.root.find(
      (inst) => inst.props && inst.props.children === 'Welcome',
    );
    expect(title).toBeTruthy();
    act(() => tr!.unmount());
  });

  it('matches snapshot (dark)', () => {
    jest.spyOn(RN, 'useColorScheme').mockReturnValue('dark');
    let tr: renderer.ReactTestRenderer;
    act(() => {
      tr = renderer.create(
        <AuthTemplate title="Login">
          <></>
        </AuthTemplate>,
      );
    });
    expect(tr!.toJSON()).toMatchSnapshot();
    act(() => tr!.unmount());
  });
});

