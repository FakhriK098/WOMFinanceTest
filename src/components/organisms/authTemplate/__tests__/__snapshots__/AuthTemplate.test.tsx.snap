// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthTemplate organism matches snapshot (dark) 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 20,
      },
      {
        "backgroundColor": "#0B1220",
      },
    ]
  }
>
  <View
    style={
      {
        "borderRadius": 16,
        "gap": 8,
        "padding": 20,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#F9FAFB",
            "fontSize": 24,
            "fontWeight": "700",
          },
          undefined,
        ]
      }
    >
      Login
    </Text>
    <View
      style={
        {
          "height": 24,
          "width": 24,
        }
      }
    />
  </View>
</View>
`;

exports[`AuthTemplate organism matches snapshot (light) and renders title/children 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 20,
      },
      {
        "backgroundColor": "#FFFFFF",
      },
    ]
  }
>
  <View
    style={
      {
        "borderRadius": 16,
        "gap": 8,
        "padding": 20,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#111827",
            "fontSize": 24,
            "fontWeight": "700",
          },
          undefined,
        ]
      }
    >
      Welcome
    </Text>
    <View
      style={
        {
          "height": 24,
          "width": 24,
        }
      }
    />
  </View>
</View>
`;
